### PR TITLE
Big scary commit

### PR DESCRIPTION
**Pull request by Protonull**

### CODE REVIEW

**Comment by Diet-Cola** [comment]
Will this break things if merged? Really not a fan of going through all plugins again to fix just utility methods

File: paper/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/ItemBuilder.java [hunk 1]
```diff
@@ -2,20 +2,23 @@
 
 import java.util.Objects;
 import java.util.function.Consumer;
-import javax.annotation.Nonnull;
+import lombok.Synchronized;
```
Comment (Maxopoly) [inline paper/src/main/java/vg/civcraft/mc/civmodcore/inventory/items/ItemBuilder.java:5]: :disappointed:

**Comment by Protonull** [comment]
> Will this break things if merged? Really not a fan of going through all plugins again to fix just utility methods

Just tested using the [current AnsibleSetup](https://github.com/CivMC/AnsibleSetup/tree/66964d2c53ad9a43791df3368cca7a03698bc23b) and there aren't any immediately obvious issues. I also used my [megarepo](https://github.com/Protonull/megarepo) tool to do searches on all CivMC's public repositories to check whether removing APIs would cause problems.

It's *possible* that problems could arise if
1. You have private repositories/plugins that I don't have access to thus couldn't check or test;
2. You have PRs that use those APIs

I'll do another round of searches just to be sure, but I don't anticipate any sudden issues arising from this PR being merged.

**Comment by Protonull** [comment]
Removed APIs:

- `ChatUtils.newComponent()` - 0 Matches
  *(Should just use `Component.empty()`)*

- `ChatUtils.cloneComponent(Component)` - 0 Matches
  *(Kyori components are immutable, so cloning means nothing.)*

- `MapColours` - 0 Matches
  *(This was a de-obfuscation class, thus is redundant with Paper's userdev)*

- `NBTTagCompoundExtensions` (replaced with `NBTCompoundUtils`) - 0 Matches
  *(Renamed to be clearer and distance from @ExtensionMethod contexts)*

- `NBTTagListExtensions` (replaced with `NBTListUtils`) - 0 Matches
  *(Renamed to be clearer and distance from @ExtensionMethod contexts)*

- `BatchedNbtStorage` - 0 Matches
  *(This has never been used and likely never will.)*

- `ParticleEffect` - 0 Matches
  *(Extremely old data class long since superceded by Paper APIs)*

- `SkinCache` (`util.SkinCache` was replaced by `utilities.SkinCache`) - 0 Matches
  *(This was moved during the Great Reorganisation.. not sure why this exists)*

- `ResourceUtils` - 0 Matches
  *(This was hardly used and now not at all since moving away from CSV files)*

- `Title` - 0 Matches
  *(Extremely old data class long since superceded by Paper APIs)*

- `ImmutableLocation` - 0 Matches
  *(Maintaining this would be an absolute pain.)*

- `ChunkOperation` - 0 Matches
  *(This has never been used and likely never will.)*

- `ChunkOperationManager` - 0 Matches
  *(This has never been used and likely never will.)*

"Matches" are determined by the number of occurances outside of internal CivModCore usage. The removed APIs and classes have been replaced. In addition, `Title` and `ParticleEffect` were old data classes that have long been superceded by Paper APIs.

**Comment by Protonull** [comment]
Any update on this?

**Comment by wingzero54** [comment]
> Any update on this?

I'll merge it when I merge in Max's, sometime this week or weekend

**Pull request closed without merging**